### ISSUE TITLE
Product API: The process of uploading images via the API is not explained clearly

### ISSUE THREAD

**Issue by dimaip** [Sylius/Sylius#10533]
**Sylius docs version**: 1.x / latest

**Description**

As I understand, currently the only way to upload product images via the API is to use form-data encoding like so:

![image](https://user-images.githubusercontent.com/837032/61428037-1f932100-a929-11e9-9ab8-287eea30c47a.png)

The docs do not mention it and instead they show a JSON example with a really cryptic note:

> The images (files) should be passed in an array as an attribute of request. See how it is done in Sylius here.

The example it leads to is a Symfony test, it's not clear for a non-Symfony developer to what request it should translate to. I also don't understand what "an attribute of request" is supposed to mean.
In any case, the only thing that works for me is form-data encoding.

**Comment by vvasiloi** [comment]
Someone please reopen this issue and do not stale.

**Comment by dimaip** [comment]
Yup, a bunch of issues I've spent hours to debug and document have been closed already. IMO stale bot is evil :-)
I'd think twice to research and open an issue for this project again, based on this experience.

**Comment by dimaip** [comment]
E.g. https://github.com/Sylius/Sylius/issues/10544

**Comment by xEndLe** [comment]
How such a basic feature does not get more attention? I don't get it. No solution since 2019? ... @dimaip

**Comment by vvasiloi** [comment]
@xEndLe You can upload files with separate requests using `Content-Type: multipart/form-data`  HTTP request header.
AFAIK this is the most practical solution.

**Comment by xEndLe** [comment]
@vvasiloi thanks for your reply.
But if we make a separate request, how do we assign this image to the specific product we want?

**Comment by vvasiloi** [comment]
@xEndLe You make a `PUT /api/v1/products/{code}` request.
